### PR TITLE
Add unit tests for db

### PR DESCRIPTION
**Pull request by claudijd**
This change is [Reviewable](https://reviewable.io/reviews/certusone/wormhole/1062)

### CODE REVIEW

File: node/pkg/db/db_test.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	assert.Nil(t, err2)
+}
+
+func TestGetSignedVAABytes(t *testing.T) {
+	dbPath := t.TempDir()
+	db, err := Open(dbPath)
+	if err != nil {
+		t.Error("failed to open database")
+	}
+	defer db.Close()
```
Comment (SEJeff) [inline node/pkg/db/db_test.go:109]: Clean up tempdir

File: node/pkg/db/db_test.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	testVaa := getVAA()
+
+	vaaID := VaaIDFromVAA(&testVaa)
+
+	privKey, _ := ecdsa.GenerateKey(crypto.S256(), rand.Reader)
+	testVaa.AddSignature(privKey, 0)
+
+	// Store full VAA
+	err2 := db.StoreSignedVAA(&testVaa)
+	assert.Nil(t, err2)
```
Comment (SEJeff) [inline node/pkg/db/db_test.go:120]:
```suggestion
	assert.NotError(t, err2)
```

File: node/pkg/db/db_test.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+	// Retrieve it using vaaID
+	vaaBytes, err2 := db.GetSignedVAABytes(*vaaID)
+	assert.Nil(t, err2)
```
Comment (SEJeff) [inline node/pkg/db/db_test.go:124]:
```suggestion
	assert.NotError(t, err2)
```

File: node/pkg/db/db_test.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+	testVaaBytes, err3 := testVaa.Marshal()
+	assert.Nil(t, err3)
```
Comment (SEJeff) [inline node/pkg/db/db_test.go:127]:
```suggestion
	assert.NotError(t, err3)
```

File: node/pkg/db/db_test.go [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+	assert.Equal(t, testVaaBytes, vaaBytes)
+}
+
+func TestFindEmitterSequenceGap(t *testing.T) {
+	dbPath := t.TempDir()
```
Comment (SEJeff) [inline node/pkg/db/db_test.go:133]: Please clean up your tempdir

File: node/pkg/db/db_test.go [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+	testVaa := getVAA()
+
+	vaaID := VaaIDFromVAA(&testVaa)
+
+	privKey, _ := ecdsa.GenerateKey(crypto.S256(), rand.Reader)
+	testVaa.AddSignature(privKey, 0)
+
+	// Store full VAA
+	err2 := db.StoreSignedVAA(&testVaa)
+	assert.Nil(t, err2)
```
Comment (SEJeff) [inline node/pkg/db/db_test.go:149]:
```suggestion
	assert.NotError(t, err2)
```

File: node/pkg/db/db_test.go [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+	resp, firstSeq, lastSeq, err := db.FindEmitterSequenceGap(*vaaID)
+
+	assert.Equal(t, []uint64{0x0}, resp)
+	assert.Equal(t, uint64(0x0), firstSeq)
+	assert.Equal(t, uint64(0x1), lastSeq)
+	assert.Nil(t, err)
```
Comment (SEJeff) [inline node/pkg/db/db_test.go:156]:
```suggestion
	assert.NotError(t, err)
```

**Review by SEJeff — DISMISSED** [review]
Using `assert.NotError()` vs `assert.Nil()` is a stylistic thing, but it makes it way more obvious what you're asserting. Not cleaning up temp stuff is just like leaving a mess in the floor then walking out. It's bad manners.

Approving because this is good stuff sans my style nits.

File: node/pkg/db/db_test.go [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+// Testing the expected default behavior of a CreateGovernanceVAA
+func TestVaaIDFromString(t *testing.T) {
+	vaaIdString := "1/0000000000000000000000000000000000000000000000000000000000000004/1"
+	vaaID, _ := VaaIDFromString(vaaIdString)
+	expectAddr := vaa.Address{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4}
+
+	assert.Equal(t, vaa.ChainIDSolana, vaaID.EmitterChain)
+	assert.Equal(t, expectAddr, vaaID.EmitterAddress)
+	assert.Equal(t, uint64(1), vaaID.Sequence)
```
[the remark this answers is not in the document]
Comment (claudijd) [reply node/pkg/db/db_test.go:41]: It feels like overkill, it hands off nearly direct to `vaa.ChainID()`, which already has sufficient coverage in `structs_test.go`.

**Review by bruce-riley — DISMISSED** [review]

**Review by claudijd — COMMENTED** [review]
Fixed.

> *[Reviewable](https://reviewable.io/reviews/certusone/wormhole/1062#-:-N5lGxoH2Jk336pQeAGO:b-lx4fjp)* status: 0 of 2 files reviewed, 11 unresolved discussions (waiting on @claudijd and @SEJeff)

**Review by claudijd — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/certusone/wormhole/1062)* status: 0 of 2 files reviewed, 11 unresolved discussions (waiting on @SEJeff)

___
*[`node/pkg/db/db_test.go` line 41 at r1](https://reviewable.io/reviews/certusone/wormhole/1062#-N-9irYg1iERq7juzYcm-r1-41:-N5lHO3ABDT0WmGsSEwc:b32w0x) ([raw file](https://github.com/certusone/wormhole/blob/fdc653be03b48e3069c0a3f47a604a7e76e14c65/node/pkg/db/db_test.go#L41)):*
<details><summary><i>Previously, claudijd (Jonathan Claudius) wrote…</i></summary><blockquote>

It feels like overkill, it hands off nearly direct to `vaa.ChainID()`, which already has sufficient coverage in `structs_test.go`.
</blockquote></details>

OK
___
*[`node/pkg/db/db_test.go` line 77 at r1](https://reviewable.io/reviews/certusone/wormhole/1062#-N-9irYg1iERq7juzYcm-r1-77:-N5lHY7t-ypQ5-V4E20V:b-896fix) ([raw file](https://github.com/certusone/wormhole/blob/fdc653be03b48e3069c0a3f47a604a7e76e14c65/node/pkg/db/db_test.go#L77)):*
<details><summary><i>Previously, SEJeff (Jeff Schroeder) wrote…</i></summary><blockquote>

You never remove the tempdir or db. You really should do that.
</blockquote></details>

Done.
___
*[`node/pkg/db/db_test.go` line 87 at r1](https://reviewable.io/reviews/certusone/wormhole/1062#-N-9irYg1iERq7juzYcm-r1-87:-N5lH_fhDapkw3rwvprH:b-896fix) ([raw file](https://github.com/certusone/wormhole/blob/fdc653be03b48e3069c0a3f47a604a7e76e14c65/node/pkg/db/db_test.go#L87)):*
<details><summary><i>Previously, SEJeff (Jeff Schroeder) wrote…</i></summary><blockquote>

Clean up the tempdir please.
</blockquote></details>

Done.
___
*[`node/pkg/db/db_test.go` line 100 at r1](https://reviewable.io/reviews/certusone/wormhole/1062#-N-9irYg1iERq7juzYcm-r1-100:-N5lHbin6bldszutR6-Z:b-896fix) ([raw file](https://github.com/certusone/wormhole/blob/fdc653be03b48e3069c0a3f47a604a7e76e14c65/node/pkg/db/db_test.go#L100)):*
<details><summary><i>Previously, SEJeff (Jeff Schroeder) wrote…</i></summary><blockquote>

```suggestion
	assert.NotError(t, err2)
```
</blockquote></details>

Done.
___
*[`node/pkg/db/db_test.go` line 109 at r1](https://reviewable.io/reviews/certusone/wormhole/1062#-N-9irYg1iERq7juzYcm-r1-109:-N5lHcgj2gqXn9PZAbp3:b-896fix) ([raw file](https://github.com/certusone/wormhole/blob/fdc653be03b48e3069c0a3f47a604a7e76e14c65/node/pkg/db/db_test.go#L109)):*
<details><summary><i>Previously, SEJeff (Jeff Schroeder) wrote…</i></summary><blockquote>

Clean up tempdir
</blockquote></details>

Done.
___
*[`node/pkg/db/db_test.go` line 120 at r1](https://reviewable.io/reviews/certusone/wormhole/1062#-N-9irYg1iERq7juzYcm-r1-120:-N5lHdDb8vzQGOb2uqOv:b-896fix) ([raw file](https://github.com/certusone/wormhole/blob/fdc653be03b48e3069c0a3f47a604a7e76e14c65/node/pkg/db/db_test.go#L120)):*
<details><summary><i>Previously, SEJeff (Jeff Schroeder) wrote…</i></summary><blockquote>

```suggestion
	assert.NotError(t, err2)
```
</blockquote></details>

Done.
___
*[`node/pkg/db/db_test.go` line 124 at r1](https://reviewable.io/reviews/certusone/wormhole/1062#-N-9irYg1iERq7juzYcm-r1-124:-N5lHdzaB6yzwmVp0LA7:b-896fix) ([raw file](https://github.com/certusone/wormhole/blob/fdc653be03b48e3069c0a3f47a604a7e76e14c65/node/pkg/db/db_test.go#L124)):*
<details><summary><i>Previously, SEJeff (Jeff Schroeder) wrote…</i></summary><blockquote>

```suggestion
	assert.NotError(t, err2)
```
</blockquote></details>

Done.
___
*[`node/pkg/db/db_test.go` line 127 at r1](https://reviewable.io/reviews/certusone/wormhole/1062#-N-9irYg1iERq7juzYcm-r1-127:-N5lHfRABgOLYPHU9rlL:b-896fix) ([raw file](https://github.com/certusone/wormhole/blob/fdc653be03b48e3069c0a3f47a604a7e76e14c65/node/pkg/db/db_test.go#L127)):*
<details><summary><i>Previously, SEJeff (Jeff Schroeder) wrote…</i></summary><blockquote>

```suggestion
	assert.NotError(t, err3)
```
</blockquote></details>

Done.
___
*[`node/pkg/db/db_test.go` line 133 at r1](https://reviewable.io/reviews/certusone/wormhole/1062#-N-9irYg1iERq7juzYcm-r1-133:-N5lHgJw5502aZhmMfwJ:b-896fix) ([raw file](https://github.com/certusone/wormhole/blob/fdc653be03b48e3069c0a3f47a604a7e76e14c65/node/pkg/db/db_test.go#L133)):*
<details><summary><i>Previously, SEJeff (Jeff Schroeder) wrote…</i></summary><blockquote>

Please clean up your tempdir
</blockquote></details>

Done.
___
*[`node/pkg/db/db_test.go` line 149 at r1](https://reviewable.io/reviews/certusone/wormhole/1062#-N-9irYg1iERq7juzYcm-r1-149:-N5lHglG4PaDEINgk-5Y:b-896fix) ([raw file](https://github.com/certusone/wormhole/blob/fdc653be03b48e3069c0a3f47a604a7e76e14c65/node/pkg/db/db_test.go#L149)):*
<details><summary><i>Previously, SEJeff (Jeff Schroeder) wrote…</i></summary><blockquote>

```suggestion
	assert.NotError(t, err2)
```
</blockquote></details>

Done.
___
*[`node/pkg/db/db_test.go` line 156 at r1](https://reviewable.io/reviews/certusone/wormhole/1062#-N-9irYg1iERq7juzYcm-r1-156:-N5lHh7C12fVhc3t8UJ2:b-896fix) ([raw file](https://github.com/certusone/wormhole/blob/fdc653be03b48e3069c0a3f47a604a7e76e14c65/node/pkg/db/db_test.go#L156)):*
<details><summary><i>Previously, SEJeff (Jeff Schroeder) wrote…</i></summary><blockquote>

```suggestion
	assert.NotError(t, err)
```
</blockquote></details>

Done.

**Review by claudijd — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/certusone/wormhole/1062#-:-N5lHzm-8w7EMr_N2kg2:bnfp4nl)* status: 0 of 2 files reviewed, 11 unresolved discussions (waiting on @SEJeff)

**Review by claudijd — COMMENTED** [review]
> Dismissed @SEJeff from 11 discussions.
> *[Reviewable](https://reviewable.io/reviews/certusone/wormhole/1062)* status: 0 of 2 files reviewed, all discussions resolved (waiting on @SEJeff)

___
*[`node/pkg/db/db_test.go` line 41 at r1](https://reviewable.io/reviews/certusone/wormhole/1062#-N-9irYg1iERq7juzYcm-r1-41:-N5lIzAR89PKbHtPRDnL:b-896fix) ([raw file](https://github.com/certusone/wormhole/blob/fdc653be03b48e3069c0a3f47a604a7e76e14c65/node/pkg/db/db_test.go#L41)):*
<details><summary><i>Previously, claudijd (Jonathan Claudius) wrote…</i></summary><blockquote>

OK
</blockquote></details>

Done.

**Review by SEJeff — APPROVED** [review]

**Review by claudijd — COMMENTED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/certusone/wormhole/1062)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @claudijd)

**Review by bruce-riley — APPROVED** [review]

**Pull request merged**